cmd/lastseen.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+func init() {
+	AddPlugin("LastSeen", "(?i)^\\.(last|seen)$", MessageHandler(LastSeen), false, false)
+}
+
+func LastSeen(msg *Message) {
+	if len(msg.Params) < 2 {
+		msg.Return("Usage: .last [nick]")
+	} else {
+		nick := msg.Params[1]
+		seen := GetLastSeen(nick)
+
+		msg.Return(fmt.Sprintf("%s was last seen %s", nick, seen))
+	}
+}
+
+func GetLastSeen(nick string) string {
+	conn := pool.Get()
+	defer conn.Close()
+
+	cmd := fmt.Sprintf("%s:%s", PREFIX, "lastseen")
+	seen, err := redis.Int64(conn.Do("hget", cmd, nick))
+
+	if err != nil {
+		return "never"
+	}
+
+	diff := time.Now().Unix() - seen
```

Comment:
Instead of worrying about doing the time calculation, just use https://github.com/dustin/go-humanize

`return humanize.Time(time.Since(time.Unix(seen, 0)))` replaces everything below this